lib/OpenQA/Worker/Cache/Service.pm

```diff
@@ -0,0 +1,206 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::Cache::Service;
+
+use Mojolicious::Lite;
+use OpenQA::Worker::Cache::Task::Asset;
+use Mojolicious::Plugin::Minion;
+use OpenQA::Worker::Cache qw(STATUS_PROCESSED STATUS_ENQUEUED STATUS_DOWNLOADING STATUS_IGNORE STATUS_ERROR);
+use Mojo::Collection;
+use Mojolicious::Plugin::Minion::Admin;
+
+use constant DEFAULT_MINION_WORKERS => 5;
+
+BEGIN { srand(time) }
+
+my $_token;
+
+sub _gen_session_token { $_token = int(rand(999999999999)) }
+
+my $enqueued = Mojo::Collection->new;
+app->hook(
+    before_server_start => sub {
+        _gen_session_token();
+        $enqueued = Mojo::Collection->new;
+    });
+
+plugin Minion => {SQLite => 'sqlite:' . OpenQA::Worker::Cache->from_worker->db_file};
+plugin 'Minion::Admin';
+plugin 'OpenQA::Worker::Cache::Task::Asset';
+plugin 'OpenQA::Worker::Cache::Task::Sync';
+
+sub SESSION_TOKEN { $_token }
+
+sub _gen_guard_name { join('.', SESSION_TOKEN, pop) }
+
+sub _exists { !!(defined $_[0] && exists $_[0]->{total} && $_[0]->{total} > 0) }
+
+sub active { !app->minion->lock(shift, 0) }
+
+sub enqueued {
+    my $lock = shift;
+    !!($enqueued->grep(sub { $_ eq $lock })->size == 1);
+}
+
+sub dequeue {
+    my $lock = shift;
+    $enqueued = $enqueued->grep(sub { $_ ne $lock });
+}
+
+sub enqueue {
+    push @$enqueued, shift;
+}
+
+sub _setup_workers {
+    return @_ unless grep { /worker/i } @_;
+
+    require OpenQA::Worker::Common;
+    my @args = @_;
+    my ($worker_settings, undef) = OpenQA::Worker::Common::read_worker_config(undef, undef);
+    my $cache_workers
+      = exists $worker_settings->{CACHEWORKERS} ? $worker_settings->{CACHEWORKERS} : DEFAULT_MINION_WORKERS;
+    push(@args, '-j', $cache_workers);
+    return @args;
+}
+
+sub run {
+    shift;
+    require OpenQA::Utils;
+    app->log->short(1);
+    OpenQA::Utils::set_listen_address(7844);
+    $ENV{MOJO_INACTIVITY_TIMEOUT} = 300;
+    require Mojolicious::Commands;
+    my @args = _setup_workers(@_);
+    app->log->debug("Starting cache service: $0 @args");
+    Mojolicious::Commands->start_app('OpenQA::Worker::Cache::Service', @args);
+}
+
+get '/session_token' => sub { shift->render(json => {session_token => SESSION_TOKEN()}) };
+
+get '/info' => sub { $_[0]->render(json => shift->minion->stats) };
+
+post '/status' => sub {
+    my $c    = shift;
+    my $data = $c->req->json;
+    my $lock = _gen_guard_name($data->{lock});
+    my $j    = $data->{id} ? app->minion->job($data->{id})->info : ();
+
+    $c->render(
+        json => {
+            status => (
+                  active($lock)   ? STATUS_DOWNLOADING
+                : enqueued($lock) ? STATUS_IGNORE
+                :                   STATUS_PROCESSED
+            ),
+            (result => $j->{result}, output => $j->{notes}->{output}) x !!($j)});
```

Comment:
This might be too clever. 😉Others working on the code will prolly understand a ternary `$j ? (...) : ()` better.

Reply:
Yes, when reviewing the code last time I also looked it up (or actually played around with it using the Perl debugger). The ternary operator would be more intuitive, indeed.

Reply:
Done in https://github.com/os-autoinst/openQA/pull/1783/commits/71e142c959efcf6c2e395f891e7522869be78d7d